### DISCUSSION TITLE
Release 7.1.0

### DISCUSSION THREAD

**Discussion by SJShaw** [antismash/antismash#673] [General]
This minor release, 7.1.0, fixes some issues identified in antiSMASH 7.0 and
adds new functionality and improvements to existing functionality.

**This release increases the minimum python version to 3.9.**

This release has a number of additions and changes to the detected cluster types.
It also updates reference databases for nearly all database comparisons and searches.

## General
* Added support for reusing JSON results compressed with bzip2
* Added error output to log files, when relevant
* Added new commandline options:
  *  `--hmmdetection-limit-to-rule-names`: restricts protoclusters detected to those types provided
  *  `--hmmdetection-limit-to-rule-categories`: restricts protoclusters detected to those of the categories provided
  *  `--html-ncbi-context`: re-enables the NCBI genomic context links which are now removed by default
* Changed all generated CDS translations to start with methionine, even when using an alternative start codon

## Detection
* Added rules for darobactin, highly reducing type II PKS, hydrogen cyanide, isocyanide, rSAM-related archaeal RiPPs, triceptide
* Added support for "extender" conditions to rules, for cases where a continguous section of some core genes might be too far from the other core genes to be properly marked
* Added extenders to trans-AT PKS rule to mark all genes in long chains of trans-AT modules as core
* Changed rule categories for larylpolyenes and PUFAs
* Changed ladderane rule to be more specific
* Changed the neighbourhood range of NI-siderophores to be wider in order to cover all of the known clusters
* Changed trans-AT PKS module classification to be less strict

## Analysis
* Removed some rare instances of fused lanthipeptide genes being interpreted as a precursors
* Updated PFam reference database from 34.0 to 35.0
* Updated GO mappings to 2023-07-05 release
* Updated ClusterBlast reference data from antiSMASH-database version 3.0 to 4.0
* Updated CompaRiPPson reference data from antiSMASH-database version 3.0 to 4.0
* Changed lanthipeptide tailoring enzyme detection to be less strict
* Changed capreomycidine substrate SMILES to reorder the backbone
* Changed all RiPP precursor ORF finding to be less strict
* Changed lasso peptide precursor handling to avoid motif duplication when shared by two protoclusters

## Output
* Added the product category of protoclusters to their relevant section in JSON results
* Added clusterblast similarity percentage to JSON results
* Removed raw ClusterBlast result file due to size and lack of clarity

## Visualisation
* Added download link for JSON results file
* Added record versions, where available, to record identifiers in HTML results
* Added a link to CDS details that allows BLAST searches against the antiSMASH-database
* Removed NCBI genomic context links by default, can be explicitly enabled with `--html-ncbi-context`
* Removed links to the now-defunct SEARCHGTr webservice
* Changed lasso peptide and sactipeptide core colouring to only cover cysteines in HTML results
* Changed CompaRiPPson characters in consensus to use the same colouring as in the queries, where relevant
* Changed default ClusterCompare comparison mode from ProtoclusterToRegion to RegionToRegion

Detailed changes since 7.0.1:
```
Kai Blin (14):
      Makefile: Add repository path to the podman image tag
      chore: Properly indent the RiPP-like rule
      hmm_rule_parser: Fix an f-string in the equivalence group handling
      hmm_detection: Add isocyanide rules
      hmm_detection: Add darobactin detection rule
      hmm_detection: Fix the category of the PKS-related arylpolyenes and PUFAs
      hmm_detection: Differentiate between real ladderanes and highly-reducing t2pks clusters
      hmm_detection: Detect hydroxycyanide biosynthesis gene clusters
      hmm_detection: Add triceptide rule
      hmm_detection: Add rule for hypothetical rSAM-related archaeal RiPPs
      chore: Add standard test headers to cyanobactin precursor tests
      lanthipepties: Make tailoring enzyme detection less strict
      hmm_detection: The rule detects hydrogen cyanide, not hydroxycyanide
      common: Support reusing from bzip2-ed json files
      chore: Bump pylint to 3.0.2 to support python 3.12

Simon Shaw (74):
      html: swap from pyScss to libsass for CSS generation
      chore: bump minimum python version to 3.9
      hmm_detection: allow categories to have parent categories
      common.serialiser: add protoclusters product category to JSON output
      html: add download link for JSON results file
      hmm_detection: increase NI-siderophore size to include all of BGC0000940
      nrps_pks_domains: widen trans-AT PKS module classification rules
      secmet: convert all CDS translations to start with methionine
      common.html_renderer: allow subsets of RiPP colouring to be specified
      common.html_renderer: allow RiPP sequences to have additional classes by position
      common.comparippson: highlight consensus characters to match other sequences
      lassopeptides: only colour cysteines in outputs
      sactipeptides: only colour cysteines in outputs
      nrps_pks: remove test data file unused since NRPSPredictor2 was removed
      nrps_pks_domains: don't clobber domain subtypes if using a mapping
      nrps_pks: pass smiles generation specific subtypes of domains
      nrps_pks: remove statement with no affect
      common.hmm_rule_parser: improve error message with empty conditions
      chore: upgrade some pylint warnings to failures
      clusterblast: add similarity percentage to JSON results
      html: use versioned record identifier in region pages
      hmm_detection: update seeds file if description file is newer
      common.hmm_rule_parser: fix out of date documentation on SUPERIORS requirements
      hmm_detection: move rule category definitions/helpers to common
      config: allow get_config() to build default values if not previously built
      config: remove unused import
      chore: remove pylint suppressions with no effect
      secmet: add a function for finding a distance between locations
      common.hmm_rule_parser: add rule extenders
      hmm_detection: use rule extenders with transAT-PKS rule
      nrps_pks: fix backbone ordering in SMILES for cap substrate
      common.serialiser: update compatible schema list
      tigrfam: use record helpers instead of a slow re-implementation
      cluster_hmmer: use record helpers instead of a slow re-implementation
      clusterblast: remove raw diamond output files
      common.signature: add member to (Hmm)Signature to track seed count
      common.hmm_rule_parser: use seeds from Signature classes instead of extra args
      common.hmm_rule_parser: add offending rule name to duplication error in parsing
      common.hmm_rule_parser: also clear parsed categories cache in test setup
      common.hmm_rule_parser: bundle rule information into a dataclass
      common.hmm_rule_parser: freeze Multiplier instances
      hmm_detection: cache default rulesets rather than rebuild for every contig in an input
      hmm_detection: add options to more specifically restrict which rules are used
      main: add html and svg output modules to timings logged
      html: only check once if TTA symbology should be shown
      secmet: cache GC content after first calculation
      secmet: allow Record to take kwargs when converting from biopython
      secmet: allow record GC content to be set in construction
      common.serialiser: keep (and reuse) GC content for records in output JSON
      download_databases: update Pfam latest to 35.0
      pfam2go: update GO mappings to 2023-07-05 release
      main: correctly log errors to logfile if specified
      smcog_trees: make pyplot handling more version-ambiguous
      chore: pin jsonschema version to avoid incompatibilities with new versions
      subprocessing.diamond: handle bugged diamond versions
      common.all_orfs: fix minimum length not being respected in all cases
      common.hmm_rule_parser: move from area formation stage to full genome
      cluster_compare: make MIBiG results visualisation default to region-to-region mode
      secmet: fix circular import
      html: change NCBI genome context links to only show when requested
      main: treat existing files in output directories as an 'input' error type
      hmm_rule_parser: fix stale documentation
      lassopeptides: fix name shadowing of submodule
      lassopeptides: avoid duplicating precursor motifs when shared by two clusters
      subprocessing: make parallel function testing more consistent
      html: remove links to defunct SEARCHGTr
      clusterblast: don't include ambiguous location markers in SVGs
      html: add a link to CDS details to blast against the antismash-db
      lanthipeptide: avoid hitting too long a gene even if it has a precursor domain
      clusterblast: update outgoing antiSMASH-DB URL for version 4.0
      download_databases: update ClusterBlast to antiSMASH-DB 4.0
      comparippson: update antiSMASH-DB link to new API endpoint
      download_databases: update compaRiPPson antiSMASH-DB to 4.0
      chore: make prerequisite and option check failures properly set exit code
      chore: bump version to 7.1.0

Zach Reitz (2):
      Explicitly state that Pfams are used for clusterhmmer and fullhmmer
      chore: add missing semicolons to secmet.scss
  ````

<hr /><em>This discussion was created from the release <a href='https://github.com/antismash/antismash/releases/tag/7-1-0'>Release 7.1.0</a>.</em>

**Comment by zdk123** [comment]
Nice work, thanks! 
What's the timeline for updated tags on dockerhub?

**Reply by kblin** [reply]
We're currently looking into an install issue with our pinned dependencies, so maybe we'll need to do another release to fix that. Linux container images will be up as soon as possible, we rely on them as well to run the web service.
